### PR TITLE
Checkout: Disable Paypal payments in the desktop app

### PR DESCRIPTION
**Pull request by Tug**
Fixes: https://github.com/Automattic/wp-calypso/issues/3410

With this change, Paypal payments are disabled in the desktop application

Before:
image

After:
image

#### Testing Instructions
- Use an account without free credits.
- In wp-desktop, move to the `calypso` folder and run `git fetch origin && git checkout fix/disable-paypal-desktop`
- Move back to wp-desktop's root and run `make run`
- In app, try to purchase a plan (or anything else), you should land on the checkout page
- Check that `or use PayPal` is not available anymore on the bottom right of the payment form.

#### Reviews
- [x] Product Review
- [x] Code Review

### CODE REVIEW

File: client/lib/cart-values/index.js [hunk 1]
```diff
@@ -123,7 +124,7 @@ function isCreditCardPaymentsEnabled( cart ) {
 }
 
 function isPayPalExpressEnabled( cart ) {
-	return cart.allowed_payment_methods.indexOf( 'WPCOM_Billing_PayPal_Express' ) >= 0;
+	return cart.allowed_payment_methods.indexOf( 'WPCOM_Billing_PayPal_Express' ) >= 0 && ! config.isEnabled( 'desktop' );
```
Comment (stephanethomas) [inline client/lib/cart-values/index.js]: I wonder if the feature flag shouldn't be the first one in the condition so we don't have to read a long expression to find out that it's behind a feature gate.

**Comment by stephanethomas** [comment]
Code looks good, I had a minor feedback but that's probably just personal preference.

File: client/lib/cart-values/index.js (hunk 1 above)
Comment (drewblaisdell) [inline client/lib/cart-values/index.js]: Why not instead add a `paypal` feature and disable it in `config/desktop.json`?

Comment (Tug) [inline client/lib/cart-values/index.js]: Agreed

Comment (stephanethomas) [inline client/lib/cart-values/index.js]: Haha, much better indeed!

**Comment by Tug** [comment]
Thanks you two! Here we go :)

It could be good to add the `"upgrades/credit-cards"` feature wall into `isCreditCardPaymentsEnabled` for consistency as well. For the moment it's only used in the purchases page and `desktop-mac-app-store` is the only config that has it disabled. I don't know enough about this config though but it looks like all payments methods are disabled for it! Maybe this is expected?

**Comment by drewblaisdell** [comment]
Code :+1:

**Comment by drewblaisdell** [comment]
Product :+1:

File: config/wpcalypso.json [hunk 2]
```diff
@@ -90,6 +90,7 @@
 		"support-user": true,
 		"upgrades/checkout": true,
 		"upgrades/credit-cards": true,
+		"upgrades/paypal": true,
```
Comment (stephanethomas) [inline config/wpcalypso.json]: Could we please keep all these properties ordered?

**Comment by fabianapsimoes** [comment]
I've updated the PR to address Stéphane's comment. Ready for review again.

**Comment by stephanethomas** [comment]
This looks good to me. Let's merge it.

**Pull request merged**